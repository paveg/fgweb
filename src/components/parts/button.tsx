
type Props = {
  apperance: 'primary' | 'secondary' | 'error' | 'neutral'
  onClick?: () => void,
}

const primaryColor = 'border-green-500 bg-green-500 hover:bg-green-600 text-white'
const secondaryColor = 'border-yellow-500 bg-yellow-500 hover:bg-yellow-600 text-white'
const errorColor = 'border-red-500 bg-red-500 hover:bg-red-600 text-white'
const neutralColor = 'border-gray-200 bg-gray-200 hover:bg-gray-300 text-black-900'

export const Button: React.FC<Props> = ({ apperance, onClick, children }): JSX.Element => {
  let color: string
  switch (apperance) {
    case 'primary':
      color = primaryColor
      break
    case "secondary":
      color = secondaryColor
      break
    case 'error':
      color = errorColor
      break
    case 'neutral':
      color = neutralColor
      break
    default:
      color = primaryColor
      break
  }

  return <button
    type="button"
    onClick={onClick}
    className={`${color} border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}>
    {children}
  </button>
}
