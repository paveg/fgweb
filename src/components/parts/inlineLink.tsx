import Link from 'next/link'

type Props = {
  href: string
}

export const InlineLink: React.FC<Props> = ({ href, children }): JSX.Element => {
  return <Link href={href}>
    <a className="underline text-blue-600 hover:text-blue-800 visited:text-purple-600">
      {children}
    </a>
  </Link >
}
