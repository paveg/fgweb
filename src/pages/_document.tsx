import Document, { Html, Head, Main, NextScript } from 'next/document';

class CustomDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html lang="ja">
        <Head>
          <meta name="Description" content="This is a Funai Gaming website that mainly describes teams and activities." />
          <link rel="dns-prefetch" href="https://www.google.co.jp" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default CustomDocument
