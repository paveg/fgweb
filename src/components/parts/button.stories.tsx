import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Button, ButtonProps } from './button';

export default {
  title: 'Example/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => {
  return <Button {...args} >Button</Button>
};

export const Default = Template.bind({});
