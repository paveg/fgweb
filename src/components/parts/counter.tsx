import { countState } from '../../states'
import { useRecoilState } from 'recoil'
import { Button } from './button'
import { InlineLink } from './inlineLink'
import React from 'react'

export const Counter = () => {
  const [count, setCount] = useRecoilState(countState)

  return <div className="mt-3">
    <p><InlineLink href="https://github.com/facebookexperimental/Recoil">Recoil</InlineLink> Sample</p><br />
    <span className="ml-3">count: {count}</span><br />
    <Button apperance="primary" onClick={() => setCount((c) => c + 1)}>count up</Button>
  </div>
}
