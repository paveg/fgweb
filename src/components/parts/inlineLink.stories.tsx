import React from 'react'
import { Story, Meta } from '@storybook/react'

import { InlineLink, InlineLinkProps } from './inlineLink'

export default {
  title: 'Example/InlineLink',
  component: InlineLink,
} as Meta

const Template: Story<InlineLinkProps> = (args) => {
  return <InlineLink {...args}>InlineLink</InlineLink>
}

export const Default = Template.bind({})
Default.args = {
  href: '/',
  target: '_blank',
}
