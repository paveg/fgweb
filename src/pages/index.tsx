import 'tailwindcss/tailwind.css'
import React from 'react';
import { Layout } from '../components/layout';
import { InlineLink } from '../components/parts/inlineLink'

export const Home: React.FC = (): JSX.Element => (
  <Layout pageTitle="Home">
    <InlineLink href={'/parts'}>Parts</InlineLink>
  </Layout>
)

export default Home
