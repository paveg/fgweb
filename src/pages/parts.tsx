import 'tailwindcss/tailwind.css'
import React from 'react';
import { Layout } from '../components/layout';
import { InlineLink } from '../components/parts/inlineLink';
import { Button } from '../components/parts/button';

export const Parts: React.FC = (): JSX.Element => (
  <Layout pageTitle="Parts" >
    <div className="w-full text-center mx-auto">
      <h2>Buttons</h2>
      <Button apperance="primary">Primary</Button>
      <Button apperance="secondary">Secondary</Button>
      <Button apperance="neutral">Neutral</Button>
      <Button apperance="error" onClick={
        () => { return window.alert('alert opened.') }
      }>Alert</Button>
    </div>
    <InlineLink href={'/'}>Home</InlineLink>
  </Layout>
)

export default Parts
