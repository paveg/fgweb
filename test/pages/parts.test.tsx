import React from 'react'
import { render, fireEvent } from '../testUtils'
import { Parts } from '../../src/pages/parts'

describe('Parts page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Parts />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Parts />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('Alert'))
    expect(window.alert).toHaveBeenCalledWith('alert opened.')
  })
})
