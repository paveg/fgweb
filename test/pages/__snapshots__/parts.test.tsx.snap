// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parts page matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex items-center justify-between flex-wrap bg-gray-900 p-6"
  >
    <div
      class="flex items-center flex-no-shrink text-white mr-6"
    >
      <span
        class="font-bold text-2xl tracking-wide"
      >
        Funai Gaming Web
      </span>
    </div>
  </nav>
  <div
    class="container mx-auto"
  >
    <main>
      <h1>
        Parts
      </h1>
      <div
        class="w-full text-center mx-auto"
      >
        <h2>
          Buttons
        </h2>
        <button
          class="border-green-500 bg-green-500 hover:bg-green-600 text-white border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          type="button"
        >
          Primary
        </button>
        <button
          class="border-yellow-500 bg-yellow-500 hover:bg-yellow-600 text-white border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          type="button"
        >
          Secondary
        </button>
        <button
          class="border-gray-200 bg-gray-200 hover:bg-gray-300 text-black-900 border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          type="button"
        >
          Neutral
        </button>
        <button
          class="border-red-500 bg-red-500 hover:bg-red-600 text-white border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
          type="button"
        >
          Alert
        </button>
      </div>
      <a
        class="underline text-blue-600 hover:text-blue-800 visited:text-purple-600"
        href="/"
      >
        Home
      </a>
      <footer />
    </main>
  </div>
</DocumentFragment>
`;
